components/earnings/commons/earnings.lua

```diff
@@ -0,0 +1,132 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Earnings
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Earnings = {}
+local MathUtils = require('Module:Math')
+local String = require('Module:StringUtils')
+local Class = require('Module:Class')
+
+---
+-- Entry point for players and individuals
+-- @player - the player/individual for whom the earnings shall be calculated
+-- @year - (optional) the year to calculate earnings for
+-- @mode - (optional) the mode to calculate earnings for
+function Earnings.calculateForPlayer(args)
+	local player = args.player
+
+    if String.isEmpty(player) then
+        return 0
+    end
+	player = mw.ext.TeamLiquidIntegration.resolve_redirect(player)
+
+    local money = 0
+    local condition = '([[players_p1::' .. player .. ']] OR'
+        .. '[[players_p2::' .. player .. ']] OR'
+        .. '[[players_p3::' .. player .. ']] OR'
+        .. '[[players_p4::' .. player .. ']] OR'
+        .. '[[players_p5::' .. player .. ']] OR'
+        .. '[[players_p6::' .. player .. ']] OR'
+        .. '[[players_p7::' .. player .. ']] OR'
+        .. '[[players_p8::' .. player .. ']] OR'
+        .. '[[players_p9::' .. player .. ']] OR'
+        .. '[[players_p10::' .. player .. ']] OR'
+        .. '[[participant::' .. player .. ']])'
+
+    money = money + Earnings._calculateIndividualEarnings(condition, args.year, args.mode)
+
+    return MathUtils._round(money)
+end
+
+function Earnings.calculateForTeam(args)
+    local team = args.team
+    if String.isEmpty(team) then
+        return 0
+    end
+	team = mw.ext.TeamLiquidIntegration.resolve_redirect(team)
+
+    local conditions = '([[participant::' .. team .. ']] OR [[extradata_participantteam::' .. team .. ']])'
+
+    if String.isNotEmpty(args.year) then
+        conditions = conditions .. ' AND [[date::>' .. args.year .. '-01-01]] AND [[date::<' .. args.year .. '-12-31]]'
+    end
+
+	if String.isNotEmpty(args.mode) then
+        conditions = conditions .. ' AND [[mode::' .. args.mode .. ']]'
+    end
+
+    local money = mw.ext.LiquipediaDB.lpdb('placement', {
+        conditions = '[[date::!1970-01-01 00:00:00]] AND [[prizemoney::>0]] AND ' .. conditions,
```

Comment:
Consider moving 1970 to the top. The default value of dates may change soon
https://discord.com/channels/93055209017729024/372075546231832576/909829455281811518